### PR TITLE
Pull non ordered material requests using supplier on purchase order

### LINKED ISSUE #8451: V8 - Purchase Order missing option to fetch items based on Supplier
In the Purchase Order, there was an option to fetch Items based on Supplier. This option is missing in version 8. In the new Purchase Order, we had button "For Supplier", but it is missing now.

https://frappe.github.io/erpnext/user/manual/en/buying/articles/pull-items-in-purchase-order-based-on-supplier6

### PR DESCRIPTION
**Pull request by rohitwaghchaure**
Added the feature to pull the non ordered material requests using supplier on purchase order
Fixed https://github.com/frappe/erpnext/issues/8451

### CODE REVIEW

File: erpnext/buying/doctype/purchase_order/purchase_order.js [hunk 1]
```diff
@@ -181,6 +181,18 @@ erpnext.buying.PurchaseOrderController = erpnext.buying.BuyingController.extend(
 				})
 			}, __("Add items from"));
 
+		cur_frm.add_custom_button(__('For Supplier'),
+			function() {
+				erpnext.utils.map_current_doc({
```
Comment (nabinhait) [inline erpnext/buying/doctype/purchase_order/purchase_order.js]: There is already a function for this, just need to bind it to the custom button
https://github.com/frappe/erpnext/blob/develop/erpnext/buying/doctype/purchase_order/purchase_order.js#L94

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/buying/doctype/purchase_order/purchase_order.js (hunk 1 above)
Comment (rohitwaghchaure) [inline erpnext/buying/doctype/purchase_order/purchase_order.js]: Fixed!

File: erpnext/buying/doctype/purchase_order/purchase_order.js [hunk 2]
```diff
@@ -94,7 +94,7 @@ erpnext.buying.PurchaseOrderController = erpnext.buying.BuyingController.extend(
 	get_items_from_open_material_requests: function() {
 		erpnext.utils.map_current_doc({
 			method: "erpnext.stock.doctype.material_request.material_request.make_purchase_order_based_on_supplier",
-			source_name: this.frm.doc.supplier,
+			source_doctype: "Supplier",
```
Comment (nabinhait) [inline erpnext/buying/doctype/purchase_order/purchase_order.js]: Why this? It needs `source_name` which is selected supplier in the PO, not `source_doctype`.

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/buying/doctype/purchase_order/purchase_order.js [hunk 3]
```diff
@@ -181,6 +182,12 @@ erpnext.buying.PurchaseOrderController = erpnext.buying.BuyingController.extend(
 				})
 			}, __("Add items from"));
 
+		cur_frm.add_custom_button(__('For Supplier'),
```
Comment (rmehta) [inline erpnext/buying/doctype/purchase_order/purchase_order.js]: `cur_frm` !!!!

Should be a button below supplier field only shown when document is not saved (__islocal)

**Review by rmehta — CHANGES_REQUESTED** [review]

**Pull request closed without merging**